token/program-2022/src/processor.rs

```diff
@@ -102,38 +102,44 @@ impl Processor {
             Rent::get()?
         };
 
-        let mut account = Account::unpack_unchecked(&new_account_info.data.borrow())?;
-        if account.is_initialized() {
-            return Err(TokenError::AlreadyInUse.into());
-        }
+        let mut account_data = new_account_info.data.borrow_mut();
+        // unpack_uninitialized checks account.base.is_initialized() under the hood
+        let mut account =
+            StateWithExtensionsMut::<Account>::unpack_uninitialized(&mut account_data)?;
 
         if !rent.is_exempt(new_account_info.lamports(), new_account_info_data_len) {
             return Err(TokenError::NotRentExempt.into());
         }
 
-        if *mint_info.key != crate::native_mint::id() {
-            let _ = Mint::unpack(&mint_info.data.borrow_mut())
-                .map_err(|_| Into::<ProgramError>::into(TokenError::InvalidMint))?;
+        // get_required_account_extensions checks mint validity
+        let required_extensions = Self::get_required_account_extensions(mint_info)?;
+        if ExtensionType::get_account_len::<Account>(&required_extensions)
+            > new_account_info_data_len
```

Comment:
What about making this `!=`? That way we're totally covered

Reply:
Sure. The `<` case should be covered in the extension initialization, but I suppose it is nicer to fail faster.